docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,121 @@
-     - Yes, responding to transactions
-     - Automatic trade registration.
-   * - Interactive
-     - Yes (and displays to user)
-     - Yes, based on user input
-     - DA’s :doc:`Navigator </tools/navigator/index>`, which lets you see and interact with the ledger
+  git clone https://github.com/digital-asset/create-daml-app
 
-Additionally, applications can be written in two different styles:
+``create-daml-app`` is a small, but fully functional demo application implementing the recommended
+architecture, including
 
--  Event-driven - applications base their actions on individual ledger events only.
--  State-driven - applications base their actions on some model of all contracts active on the ledger.
+- DAML React libraries
+- quick iteration against the :ref:`DAML Ledger Sandbox <sandbox-manual>`.
+- authentication
+- deploy your application in the cloud as a Docker container
 
-Event-driven applications
-=========================
+It provides an excellent starting point to iterate on, so you can build your own application.
 
-**Event-driven** applications react to events on the the ledger and generate commands and other outputs on a per-event basis. They do not require access to ledger state beyond the event they are reacting to.
 
-Examples are sink applications that read the ledger and dump events to an external store (e.g. an external (reporting) database).
+Frontend
+~~~~~~~~
 
-State-driven applications
-=========================
+We recommended building your frontend with the `React <https://reactjs.org>`_ framework. We provide
+several support libraries to build your React frontend for a DAML application.
 
-**State-driven** applications build up a real-time view of the ledger state by reading events and recording contract create and archive events. They then generate commands based on a given state, not just single events.
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| Name                                                         | Summary                                                                  |
++==============================================================+==========================================================================+
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
```

Comment:
Maybe “Typescript equivalents for DAML data types” or something like that would be clearer.

Reply:
Another option "DAML data types represented in TypeScript"

Reply:
Also update the description at npmjs.com to something like "Library for generating DAML data types in TypeScript. If you are writing a full-stack DAML application you will need it in order to be able to communicate between your React components and DAML code"

Would be good to explain there why this is needed

Reply:
@daml/types is not a library to generate data types in typescript, it's equivalents of core DAML types in typescript. Maybe it shouldn't be listed here at all because it's more of an implementation detail of the other two libraries. It would rather belong to a page describing our javascript bindings for people that want to use a different javascript frontend framework than react.

Reply:
If it's a detail and users don't used it then you're absolutely right: they don't need to know about it 😄As we mention it later

"If you choose a different Javascript based frontend framework, the packages ``@daml/ledger``,
``@daml/types`` and the generated ``@daml2ts`` package provide you with the necessary interface code to connect and issue commands against your ledger."

a short clarification at npmjs.com would be good enough. Or we just don't mention it in the referenced sentence 😄

Reply:
I removed it and will reintroduce it in a separate javascript bindings page.